### PR TITLE
Kanban: fix columns ordering in "Add column" list

### PR DESCRIPTION
**Pull request by AdrienClairembault**
Before:

image

After:

image

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #28851

### CODE REVIEW

**Comment by trasher** [comment]
> I suppose the server could return a simple sorted array instead of an associative one.

A reason that should not be done now?

**Comment by AdrienClairembault** [comment]
> > I suppose the server could return a simple sorted array instead of an associative one.
> 
> A reason that should not be done now?

More code change as il would impact some other js functions that need to be updated (= higher risk of breaking something for a feature that is not functionally tested).

I think its better to do it in the planned rewrite targeting `main`

**Comment by trasher** [comment]
OK

**Review by orthagh — APPROVED** [review]

File: js/modules/Kanban/Kanban.js [hunk 1]
```diff
@@ -1047,13 +1047,34 @@ class GLPIKanbanRights {
...
+                    const name_a = a.name.toUpperCase();
+                    const name_b = b.name.toUpperCase();
+                    if (name_a < name_b) {
+                        return -1;
+                    } else if (name_a > name_b) {
+                        return 1;
+                    } else {
+                        return 0;
+                    }
+                });
```
Comment (cedric-anne) [inline js/modules/Kanban/Kanban.js:1066]: If you want to have a natural sorting, you should probably use something like:
```js
const collator = new Intl.Collator(undefined, {numeric: true, sensitivity: 'base'});
sorted_data.sort(collator.compare);
```

**Review by cedric-anne — APPROVED** [review]

**Review by cconard96 — APPROVED** [review]

**Pull request merged**